### PR TITLE
feat:[CDS-70052]: Serverless Prepare Rollback and Rollback Manager Side Changes

### PR DESCRIPTION
**Pull request by piyush-bhuwalka**
## Describe your changes

Serverless 2.0 Prepare Rollback and Rollback Manager Side Changes 

Includes :

- Skeleton Changes
- Logic wise changes 
- Test cases

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/48946)

### CODE REVIEW

**Comment by piyush-bhuwalka** [comment]
trigger smartchecks

**Comment by piyush-bhuwalka** [comment]
trigger uts

**Comment by piyush-bhuwalka** [comment]
trigger uts

**Comment by piyush-bhuwalka** [comment]
trigger checkstylecodeformat

**Comment by piyush-bhuwalka** [comment]
trigger messagecheck

**Comment by piyush-bhuwalka** [comment]
trigger uts

File: 127-cd-nextgen-entities/src/main/java/io/harness/cdng/visitor/YamlTypes.java [hunk 1]
```diff
@@ -227,4 +227,5 @@ public interface YamlTypes {
 
   // SERVERLESS CONTAINER STEPS
   String SERVERLESS_PREPARE_ROLLBACK = StepSpecTypeConstants.SERVERLESS_PREPARE_ROLLBACK;
```
Comment (sainath-b) [inline 127-cd-nextgen-entities/src/main/java/io/harness/cdng/visitor/YamlTypes.java:229]: Name this something like SERVERLESS_AWS_LAMBDA_ROLLBACK_V2

File: 127-cd-nextgen-entities/src/main/java/io/harness/executions/steps/ExecutionNodeType.java [hunk 2]
```diff
@@ -149,7 +149,8 @@ public enum ExecutionNodeType {
       "GOOGLE_CLOUD_FUNCTIONS_GEN_ONE_ROLLBACK", YamlTypes.GOOGLE_CLOUD_FUNCTIONS_GEN_ONE_ROLLBACK),
   SERVICE_HOOKS("SERVICE_HOOKS", YamlTypes.SERVICE_HOOKS),
   K8S_BLUE_GREEN_STAGE_SCALE_DOWN("K8S_BLUE_GREEN_STAGE_SCALE_DOWN", YamlTypes.K8S_BLUE_GREEN_STAGE_SCALE_DOWN),
-  SERVERLESS_PREPARE_ROLLBACK("SERVERLESS_PREPARE_ROLLBACK", YamlTypes.SERVERLESS_PREPARE_ROLLBACK);
```
Comment (sainath-b) [inline 127-cd-nextgen-entities/src/main/java/io/harness/executions/steps/ExecutionNodeType.java:152]: Name this something like SERVERLESS_AWS_LAMBDA_PREPARE_ROLLBACK_V2. Other constants as well.

**Comment by piyush-bhuwalka** [comment]
trigger uts

File: 127-cd-nextgen-entities/src/main/java/io/harness/executions/steps/ExecutionNodeType.java (hunk 2 above)
Comment (piyush-bhuwalka) [reply 127-cd-nextgen-entities/src/main/java/io/harness/executions/steps/ExecutionNodeType.java:152]: Done

File: 127-cd-nextgen-entities/src/main/java/io/harness/cdng/visitor/YamlTypes.java (hunk 1 above)
Comment (piyush-bhuwalka) [reply 127-cd-nextgen-entities/src/main/java/io/harness/cdng/visitor/YamlTypes.java]: Done

**Comment by yogesh-chauhan** [comment]
*[`125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/pipeline/steptype/NGStepType.java` line 312 at r5](https://reviewable.io/reviews/harness/harness-core/48946#-NXTWic21sWzDnQTU3xh:-NXTWic21sWzDnQTU3xi:b-v1at1z) ([raw file](https://github.com/harness/harness-core/blob/6ddcd8ead913b20f1e376cffe8691253b29c5f7d/125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/pipeline/steptype/NGStepType.java#L312)):*
> ```Java
>       StepSpecTypeConstants.K8S_BLUE_GREEN_STAGE_SCALE_DOWN),
>   @JsonProperty(StepSpecTypeConstants.SERVERLESS_AWS_LAMBDA_PREPARE_ROLLBACK_V2)
>   SERVERLESS_AWS_LAMBDA_PREPARE_ROLLBACK_V2("Serverless Prepare Rollback",
> ```

is this already used in production ?

**Review by yogesh-chauhan — COMMENTED** [review]
> Reviewed 1 of 51 files at r1, 1 of 42 files at r2, 6 of 23 files at r4.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/48946)* status: 59 file(s) to review. 2. unresolved discussions. 0 of 2 approvals obtained.

**Comment by piyush-bhuwalka** [comment]
*[`125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/pipeline/steptype/NGStepType.java` line 312 at r5](https://reviewable.io/reviews/harness/harness-core/48946#-NXTWic21sWzDnQTU3xh:-NXTYMP-4TFe4mONBW5s:be49kz3) ([raw file](https://github.com/harness/harness-core/blob/6ddcd8ead913b20f1e376cffe8691253b29c5f7d/125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/pipeline/steptype/NGStepType.java#L312)):*
<details><summary><i>Previously, yogesh-chauhan (Yogesh Chauhan) wrote…</i></summary><blockquote>

is this already used in production ?
</blockquote></details>

No it has not gone to production. We merged it yesterday and it is not even in pre_qa/portal

File: 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/serverless/request/ServerlessCloudFormationRollbackRequest.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    List<EncryptedDataDetail> cloudProviderEncryptionDetails = serverlessInfra.getEncryptionDataDetails();
+    List<ExecutionCapability> capabilities =
+        new ArrayList<>(EncryptedDataDetailsCapabilityHelper.fetchExecutionCapabilitiesForEncryptedDataDetails(
+            cloudProviderEncryptionDetails, maskingEvaluator));
+    if (serverlessInfra instanceof ServerlessAwsLambdaInfraConfig) {
+      AwsConnectorDTO awsConnectorDTO = ((ServerlessAwsLambdaInfraConfig) serverlessInfra).getAwsConnectorDTO();
+      capabilities.addAll(AwsCapabilityHelper.fetchRequiredExecutionCapabilities(awsConnectorDTO, maskingEvaluator));
+      CrossAccountAccessDTO crossAccountAccess = awsConnectorDTO.getCredential().getCrossAccountAccess();
+      if (crossAccountAccess != null && crossAccountAccess.getCrossAccountRoleArn() != null) {
+        capabilities.add(AwsCliInstallationCapability.builder().criteria("AWS Cli Installed").build());
```
Comment (sainath-b) [inline 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/serverless/request/ServerlessCloudFormationRollbackRequest.java:63]: why do we need these capability?

File: 125-cd-nextgen/src/test/resources/schema/DeploymentStage/all.json [hunk 4]
```diff
@@ -18326,14 +18370,14 @@
...
-                "const": "ServerlessPrepareRollback"
+                "const": "ServerlessAwsLambdaRollbackV2"
               }
             }
           },
           "then": {
             "properties": {
               "spec": {
-                "$ref": "#/definitions/ServerlessAwsLambdaPrepareRollbackContainerStepInfo"
+                "$ref": "#/definitions/ServerlessAwsLambdaCloudFormationRollbackStepInfo"
```
Comment (sainath-b) [inline 125-cd-nextgen/src/test/resources/schema/DeploymentStage/all.json:18380]: Lets Not call it cloud formation rollback step info. We can follow same standard like serverless aws lambda rollback v2 step info. Same at other places as well.

File: 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/serverless/request/ServerlessCloudFormationRollbackRequest.java (hunk 3 above)
Comment (piyush-bhuwalka) [reply 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/serverless/request/ServerlessCloudFormationRollbackRequest.java]: Removed

File: 125-cd-nextgen/src/test/resources/schema/DeploymentStage/all.json (hunk 4 above)
Comment (piyush-bhuwalka) [reply 125-cd-nextgen/src/test/resources/schema/DeploymentStage/all.json]: Done

File: 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/pipeline/steptype/NGStepType.java [hunk 5]
```diff
@@ -310,7 +310,15 @@ public enum NGStepType {
   @JsonProperty(StepSpecTypeConstants.K8S_BLUE_GREEN_STAGE_SCALE_DOWN)
   BLUE_GREEN_STAGE_SCALE_DOWN("Blue Green Stage Scale Down",
       Collections.singletonList(ServiceDefinitionType.KUBERNETES), "Kubernetes",
-      StepSpecTypeConstants.K8S_BLUE_GREEN_STAGE_SCALE_DOWN);
+      StepSpecTypeConstants.K8S_BLUE_GREEN_STAGE_SCALE_DOWN),
+  @JsonProperty(StepSpecTypeConstants.SERVERLESS_AWS_LAMBDA_PREPARE_ROLLBACK_V2)
+  SERVERLESS_AWS_LAMBDA_PREPARE_ROLLBACK_V2("Serverless Prepare Rollback",
```
Comment (piyush-bhuwalka) [inline 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/pipeline/steptype/NGStepType.java:315]: name change

File: 125-cd-nextgen/src/main/java/io/harness/cdng/pipeline/CDStepInfo.java [hunk 6]
```diff
@@ -137,7 +139,8 @@
         GoogleFunctionsTrafficShiftStepInfo.class, GoogleFunctionsRollbackStepInfo.class, AwsSamDeployStepInfo.class,
         AwsSamBuildStepInfo.class, AwsLambdaDeployStepInfo.class, AwsSamRollbackStepInfo.class,
         AwsLambdaRollbackStepInfo.class, BambooBuildStepInfo.class, TasRouteMappingStepInfo.class,
-        GoogleFunctionsGenOneDeployStep.class, GoogleFunctionsGenOneRollbackStep.class, K8sBGStageScaleDownStep.class})
+        GoogleFunctionsGenOneDeployStep.class, GoogleFunctionsGenOneRollbackStep.class, K8sBGStageScaleDownStep.class,
+        ServerlessAwsLambdaPrepareRollbackV2Step.class, ServerlessAwsLambdaRollbackV2Step.class})
```
Comment (piyush-bhuwalka) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/pipeline/CDStepInfo.java:143]: use stepInfo

File: 125-cd-nextgen/src/main/java/io/harness/cdng/serverless/container/steps/ServerlessAwsLambdaPrepareRollbackV2Step.java [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+      } catch (Exception e) {
+        log.error("Error while parsing Stack Details", e);
+      }
+
+      ServerlessAwsLambdaPrepareRollbackDataOutcome serverlessAwsLambdaPrepareRollbackDataOutcome = null;
+      if (stackDetails != null) {
+        serverlessAwsLambdaPrepareRollbackDataOutcome =
+            ServerlessAwsLambdaPrepareRollbackDataOutcome.builder().stackDetails(stackDetails).build();
+        executionSweepingOutputService.consume(ambiance,
+            OutcomeExpressionConstants.SERVERLESS_AWS_LAMBDA_PREPARE_ROLLBACK_DATA_OUTCOME,
```
Comment (piyush-bhuwalka) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/serverless/container/steps/ServerlessAwsLambdaPrepareRollbackV2Step.java:138]: check v2 should be in outcome name

File: 125-cd-nextgen/src/main/java/io/harness/cdng/serverless/container/steps/ServerlessAwsLambdaRollbackV2Step.java [hunk 8]
```diff
@@ -0,0 +1,173 @@
...
+  @Override
+  public TaskRequest obtainTaskAfterRbac(
+      Ambiance ambiance, StepElementParameters stepElementParameters, StepInputPackage inputPackage) {
+    ServerlessAwsLambdaRollbackV2StepParameters rollbackStepParameters =
+        (ServerlessAwsLambdaRollbackV2StepParameters) stepElementParameters.getSpec();
+    if (EmptyPredicate.isEmpty(rollbackStepParameters.getServerlessAwsLambdaRollbackFnq())) {
+      return TaskRequest.newBuilder()
+          .setSkipTaskRequest(
+              SkipTaskRequest.newBuilder()
+                  .setMessage("Serverless Aws Lambda Prepare Rollback step was not executed. Skipping rollback.")
```
Comment (piyush-bhuwalka) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/serverless/container/steps/ServerlessAwsLambdaRollbackV2Step.java:88]: v2 in the msg

File: 125-cd-nextgen/src/main/java/io/harness/cdng/serverless/container/steps/ServerlessAwsLambdaRollbackV2Step.java [hunk 9]
```diff
@@ -0,0 +1,173 @@
...
+    }
+    OptionalSweepingOutput serverlessRollbackDataOptionalOutput =
+        executionSweepingOutputService.resolveOptional(ambiance,
+            RefObjectUtils.getSweepingOutputRefObject(rollbackStepParameters.getServerlessAwsLambdaRollbackFnq() + "."
+                + OutcomeExpressionConstants.SERVERLESS_AWS_LAMBDA_PREPARE_ROLLBACK_DATA_OUTCOME));
+    if (!serverlessRollbackDataOptionalOutput.isFound()) {
+      return TaskRequest.newBuilder()
+          .setSkipTaskRequest(
+              SkipTaskRequest.newBuilder()
+                  .setMessage("Serverless Aws Lambda Prepare Rollback step was not executed. Skipping rollback.")
```
Comment (piyush-bhuwalka) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/serverless/container/steps/ServerlessAwsLambdaRollbackV2Step.java:100]: v2 in the msg

File: 125-cd-nextgen/src/main/java/io/harness/cdng/serverless/container/steps/ServerlessAwsLambdaRollbackV2Step.java [hunk 10]
```diff
@@ -0,0 +1,173 @@
...
+      ThrowingSupplier<ServerlessCommandResponse> responseDataSupplier) throws Exception {
+    StepResponse stepResponse = null;
+    try {
+      ServerlessRollbackResponse rollbackResponse = (ServerlessRollbackResponse) responseDataSupplier.get();
+      StepResponseBuilder stepResponseBuilder =
+          StepResponse.builder().unitProgressList(rollbackResponse.getUnitProgressData().getUnitProgresses());
+
+      stepResponse = generateStepResponse(ambiance, rollbackResponse, stepResponseBuilder);
+    } catch (Exception e) {
+      log.error("Error while processing Serverless Aws Lambda rollback response: {}", ExceptionUtils.getMessage(e), e);
```
Comment (piyush-bhuwalka) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/serverless/container/steps/ServerlessAwsLambdaRollbackV2Step.java:143]: v2 in the msg

File: 125-cd-nextgen/src/main/java/io/harness/serializer/CDNGRegistrars.java [hunk 11]
```diff
@@ -1260,5 +1262,29 @@ public class CDNGRegistrars {
...
                    .build())
+          .add(YamlSchemaRootClass.builder()
+                   .entityType(EntityType.SERVERLESS_AWS_LAMBDA_PREPARE_ROLLBACK_V2)
+                   .availableAtProjectLevel(true)
+                   .availableAtOrgLevel(false)
+                   .availableAtAccountLevel(false)
+                   .clazz(ServerlessAwsLambdaPrepareRollbackV2StepNode.class)
+                   .yamlSchemaMetadata(YamlSchemaMetadata.builder()
+                                           .namespace(SchemaNamespaceConstants.CD)
+                                           .modulesSupported(Arrays.asList(ModuleType.CD, ModuleType.PMS))
```
Comment (piyush-bhuwalka) [inline 125-cd-nextgen/src/main/java/io/harness/serializer/CDNGRegistrars.java:1273]: check if pms is required

File: 950-delegate-tasks-beans/src/main/java/software/wings/beans/TaskType.java [hunk 12]
```diff
@@ -487,7 +487,8 @@ public enum TaskType {
   RANCHER_TEST_CONNECTION_TASK_NG(TaskGroup.K8S_NG, "Rancher connector test connection task for NG"),
   RANCHER_LIST_CLUSTERS_TASK_NG(TaskGroup.K8S_NG, "Rancher list clusters task for NG"),
   COMMAND_TASK_NG_WITH_OUTPUT_VARIABLE_SECRETS(TaskGroup.COMMAND_TASK_NG, "Command Task"),
-  NOTIFY_WEBHOOK(TaskGroup.NOTIFICATION);
+  NOTIFY_WEBHOOK(TaskGroup.NOTIFICATION),
+  SERVERLESS_ROLLBACK_V2_TASK(TaskGroup.SERVERLESS_NG, "Serverless CloudFormation Rollback Task");
```
Comment (piyush-bhuwalka) [inline 950-delegate-tasks-beans/src/main/java/software/wings/beans/TaskType.java:491]: name change

File: 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/pipeline/steptype/NGStepType.java (hunk 5 above)
Comment (piyush-bhuwalka) [reply 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/pipeline/steptype/NGStepType.java:315]: Done

File: 125-cd-nextgen/src/main/java/io/harness/cdng/creator/CDNGPlanCreatorProvider.java [hunk 13]
```diff
@@ -1386,6 +1398,27 @@ public List<StepInfo> getStepInfo() {
             .setFeatureFlag(FeatureName.CDS_BG_STAGE_SCALE_DOWN_STEP_NG.name())
             .build();
 
+    StepInfo serverlessAwsLambdaPrepareRollbackV2 =
+        StepInfo.newBuilder()
+            .setName("Serverless Aws Lambda Prepare Rollback V2")
+            .setType(StepSpecTypeConstants.SERVERLESS_AWS_LAMBDA_PREPARE_ROLLBACK_V2)
+            .setStepMetaData(StepMetaData.newBuilder()
+                                 .addCategory("SERVERLESS_AWS_LAMBDA_PREPARE_ROLLBACK_V2")
+                                 .setFolderPath("Serverless Lambda")
```
[the remark this answers is not in the document]
Comment (piyush-bhuwalka) [reply 125-cd-nextgen/src/main/java/io/harness/cdng/creator/CDNGPlanCreatorProvider.java:1407]: Keeping it same as v1

File: 125-cd-nextgen/src/main/java/io/harness/cdng/pipeline/CDStepInfo.java (hunk 6 above)
Comment (piyush-bhuwalka) [reply 125-cd-nextgen/src/main/java/io/harness/cdng/pipeline/CDStepInfo.java:143]: Done

File: 125-cd-nextgen/src/main/java/io/harness/cdng/serverless/container/steps/ServerlessAwsLambdaPrepareRollbackV2Step.java (hunk 7 above)
Comment (piyush-bhuwalka) [reply 125-cd-nextgen/src/main/java/io/harness/cdng/serverless/container/steps/ServerlessAwsLambdaPrepareRollbackV2Step.java:138]: Done

File: 125-cd-nextgen/src/main/java/io/harness/cdng/serverless/container/steps/ServerlessAwsLambdaRollbackV2Step.java (hunk 9 above)
Comment (piyush-bhuwalka) [reply 125-cd-nextgen/src/main/java/io/harness/cdng/serverless/container/steps/ServerlessAwsLambdaRollbackV2Step.java:100]: Done

File: 125-cd-nextgen/src/main/java/io/harness/cdng/serverless/container/steps/ServerlessAwsLambdaRollbackV2Step.java (hunk 8 above)
Comment (piyush-bhuwalka) [reply 125-cd-nextgen/src/main/java/io/harness/cdng/serverless/container/steps/ServerlessAwsLambdaRollbackV2Step.java:88]: Done

File: 125-cd-nextgen/src/main/java/io/harness/cdng/serverless/container/steps/ServerlessAwsLambdaRollbackV2StepNode.java [hunk 14]
```diff
@@ -0,0 +1,60 @@
...
+@NoArgsConstructor
+@EqualsAndHashCode(callSuper = true)
+@JsonTypeName(StepSpecTypeConstants.SERVERLESS_AWS_LAMBDA_ROLLBACK_V2)
+@TypeAlias("serverlessAwsLambdaRollbackV2StepNode")
+@RecasterAlias("io.harness.cdng.serverless.container.steps.ServerlessAwsLambdaRollbackV2StepNode")
+public class ServerlessAwsLambdaRollbackV2StepNode extends CdAbstractStepNode {
+  @JsonProperty("type") @NotNull StepType type = StepType.ServerlessAwsLambdaRollbackV2;
+  @JsonProperty("spec")
+  @JsonTypeInfo(use = NAME, property = "type", include = EXTERNAL_PROPERTY, visible = true)
+  ServerlessAwsLambdaRollbackV2StepInfo serverlessAwsLambdaCloudFormationRollbackStepInfo;
```
[the remark this answers is not in the document]
Comment (piyush-bhuwalka) [reply 125-cd-nextgen/src/main/java/io/harness/cdng/serverless/container/steps/ServerlessAwsLambdaRollbackV2StepNode.java:41]: Done

File: 950-delegate-tasks-beans/src/main/java/software/wings/beans/TaskType.java (hunk 12 above)
Comment (piyush-bhuwalka) [reply 950-delegate-tasks-beans/src/main/java/software/wings/beans/TaskType.java:491]: Done

**Comment by piyush-bhuwalka** [comment]
trigger uts

**Comment by piyush-bhuwalka** [comment]
trigger uts

**Comment by piyush-bhuwalka** [comment]
trigger uts

**Comment by piyush-bhuwalka** [comment]
trigger messagecheck

**Review by RishabhGupta34 — APPROVED** [review]

**Review by yogesh-chauhan — APPROVED** [review]
> Reviewed 3 of 51 files at r1, 12 of 42 files at r2, 1 of 4 files at r3, 23 of 47 files at r6, 2 of 7 files at r7, 21 of 22 files at r8, 1 of 1 files at r9, 1 of 1 files at r10, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/48946)* status: 64 file(s) to review. 4. unresolved discussions.

**Comment by piyush-bhuwalka** [comment]
trigger review

**Pull request merged**